### DISCUSSION TITLE
Lastets update broke eslint "could not find config file"

### DISCUSSION THREAD

**Discussion by sample-usr** [LazyVim/LazyVim#4108] [Q&A] [answered]
I updated lazyvim yesterday and now I'm constantly getting errors on all typescript/tsx files.
```
eslint: Request text/Document diagnostic failed with message: Could not find config file
```
I do have a config file `.eslintrc.js` at the root of my project and everything was working fine till yesterday. Anyone has a clue? I looked at the changelog from yesterday but doesn't seem like anything was changed regarding eslint or linting.

**Comment by folke** [comment, marked as answer]
something to do with the new eslint config format.
The old format is deprecated.
Install an old version of eslint, or update your config.
Not a LazyVim issue

**Reply by sample-usr** [reply]
Thanks, that was it, I was still using the old config format, downgraded eslint to `8.56.0` and that fixed the issue.

**Reply by sajidali2444** [reply]
exactly this solution worked when I downgraded to eslint 8.56.0 problem resolved

**Reply by schachmatthias** [reply]
Downgrading may resolve the issue but is def not a solution just a workaround. You usually want to have the newest version installed. Eslint has quiet a few breaking changes from ver 8 to 9.

**Comment by dpetka2001** [comment]
Take a look at #2732, which seems to also deal with the same issue and supposedly has a suggested answer that works (I'm saying `supposedly` because I don't use it myself and can't test it properly).

**Reply by sample-usr** [reply]
Thanks, I think I just need to update to the new flat file config format from eslint.

**Comment by jrock2004** [comment]
So it looks like eslint is no longer going to [support eslint config in package json](https://github.com/eslint/eslint/discussions/18131) anymore. Did I read this wrong?

**Reply by sample-usr** [reply]
Yes, also I recently took some time to upgrade to eslint 9 but gave up after spending 3 hours just trying to get the same configuration that I had previously. I have since moved on to [biome](https://biomejs.dev/), just a lot simpler and doesn't get in the way.

**Comment by LukaHedtSV** [comment]
For anyone who chances upon this issue in future, and tooling prevents you from migrating config file setups (Thanks `projen` :/), if you open neovim with the env var `ESLINT_USE_FLAT_CONFIG=false` in your system, it will run normally (at least as far as eslint 9 would run using the old config file style)

For me, I just had to update my shell alias to literally be:

```sh
# ~/.zshrc
alias vim="ESLINT_USE_FLAT_CONFIG=false nvim"
```

And now opening directories like `vim .` works fine.

**Comment by NielsCodes** [comment]
I'm not a LazyVim user, but I've been getting the same error recently and this thread is the first result when Googling the error.
I solved it by installing `@eslint/js` as a dev dependency in my app and restarting Neovim. Hope it can help someone else too.
